### PR TITLE
[build] Refactoring of config lib and ocamldebug tweaks.

### PR DESCRIPTION
**Pull request by ejgallego**
We make `config` into a proper library. This is more uniform and
useful for some clients. This also matches what was done in Dune.

Next step would be to push dependencies on `Coq_config` upwards, only
the actual toplevel binaries should depend on it.

We also remove the stale `camlp5.dbg` and refactor the dbg files a
bit, isolating the bits that are specific to the plugin / lib building
method used by makefile.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -261,7 +261,7 @@ cacheclean:
 	find theories plugins test-suite -name '.*.aux' -exec rm -f {} +
 
 cleanconfig:
-	rm -f config/Makefile config/coq_config.ml myocamlbuild_config.ml dev/ocamldebug-coq dev/camlp5.dbg config/Info-*.plist
+	rm -f config/Makefile config/coq_config.ml myocamlbuild_config.ml dev/ocamldebug-coq config/Info-*.plist
```
Comment (SkySkimmer) [inline Makefile]: Do we still use this myocamlbuild thing by the way?

**Review by SkySkimmer — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (ejgallego) [reply Makefile]: > Do we still use this myocamlbuild thing by the way?

#8754

**Comment by ejgallego** [comment]
Rebasing to solve a conflict, who would like to be the assignee?

**Review by ppedrot — COMMENTED** [review]
Not skilled to review this PR...

**Comment by ejgallego** [comment]
Maybe @SkySkimmer can be the assignee here?

**Comment by SkySkimmer** [comment]
Still waiting for @gares review.

**Comment by SkySkimmer** [comment]
@gares if you don't reply today I'll assume you don't care about this PR and I'll merge it.

**Pull request merged**